packages/mobile/src/screens/chat-screen/ChatScreen.tsx

```diff
@@ -302,9 +323,75 @@ export const ChatScreen = () => {
     )
   }
 
+  const closeReactionPopup = useCallback(() => {
+    setShouldShowPopup(false)
+    setPopupChatIndex(null)
+    setSelectedReaction('')
+  }, [setShouldShowPopup, setPopupChatIndex])
+
+  const handleReactionSelected = useCallback(
+    (message: Nullable<ChatMessage>, reaction: ReactionTypes) => {
+      if (userId && message) {
+        dispatch(
+          setMessageReaction({
+            userId,
+            chatId,
+            messageId: message.message_id,
+            reaction:
+              message.reactions?.find((r) => r.user_id === userIdEncoded)
+                ?.reaction === reaction
+                ? null
+                : reaction
+          })
+        )
+      }
+      closeReactionPopup()
+    },
+    [dispatch, userIdEncoded, chatId, userId, closeReactionPopup]
+  )
+
+  // Calculates positions of reactions popup and highlighted chat message to display
+  // in the portal.
+  const handleMessagePress = async (index: number) => {
+    if (index < 0 || index >= chatMessages.length) {
+      return
+    }
+    const popupChatMessage = chatMessages[index]
+    const popupViewRef = itemsRef.current[index]
+    if (popupViewRef === null || popupViewRef === undefined) {
+      return
+    }
+    setPopupChatIndex(index)
+    reactionInfo.current.hasTail = hasTail(
+      popupChatMessage,
+      chatMessages[index - 1]
+    )
+    const senderUserId = decodeHashId(popupChatMessage.sender_user_id)
+    reactionInfo.current.isAuthor = senderUserId === userId
+    const selectedReactionValue = popupChatMessage.reactions?.find(
+      (r) => r.user_id === encodeHashId(userId)
+    )?.reaction
+    if (selectedReactionValue) {
+      setSelectedReaction(selectedReactionValue)
```

Comment:
This doesn't feel like it should be a state member of the `ChatScreen`? Maybe a state member of the `ReactionPopup` with an initial value set from here instead?

Reply:
Don't think it even needs to be a state variable once it's inside the popup!